### PR TITLE
Add tests for flag correctness

### PR DESCRIPTION
**Pull request by tempoz**
Add tests to ensure newly added flags which contain a `.` are all known to the yaml config.

---

**Version bump**: None

### CODE REVIEW

File: enterprise/server/cmd/executor/main_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package main
```
[the remark this answers is not in the document]
Comment (tempoz) [reply enterprise/server/cmd/executor/main_test.go:1]: Good idea, I'll try it.

[the remark this answers is not in the document]
Comment (tempoz) [reply enterprise/server/cmd/executor/main_test.go:1]: Unfortunately, some top-level flags started stepping on each other; I still took the functionality and jammed it in the testing flags lib, though, so at least we're not repeating ourselves three times.

**Review by tylerwilliams — APPROVED** [review]
(1 nit)

File: server/util/testing/flags/flags.go [hunk 2]
```diff
@@ -23,3 +27,21 @@ func Set(t testing.TB, name, value string) {
 		config.TestOnlySetFlag(name, wasSet)
 	})
 }
+
+func CheckFlagsAgainstConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tempoz) [reply server/util/testing/flags/flags.go:31]:
```suggestion
// CheckFlagsAgainstConfig checks that all defined flags containing a `.` are present in the config struct.
func CheckFlagsAgainstConfig(t *testing.T) {
```

**Pull request merged**